src/ConstructiveSolidGeometry/Sampling.jl

```diff
@@ -0,0 +1,11 @@
+function sample_surface(c::AbstractVolumePrimitive{T}, sampling::Union{Real, NTuple{3,Int}}) where {T}
+    samples = CylindricalPoint{T}[]
```

Comment:
do we want this to be `CylindricalPoint{T}` by default for any Volume Primitive?
Does that work if you sample a `Box{T}` for example?

Reply:
We actually should not use push here at all. 
We can calculate the number of sample points and can allocated the memory of the corresponding vector.

Reply:
And yes, Cylindrical and Cartesian depends on the primitive.